File: backend/app/schemas/medecin.py
from datetime import datetime

from pydantic import BaseModel, EmailStr, Field


class MedecinSchema(BaseModel):
    id: int
    email: EmailStr
    nom: str
    actif: bool
    created_at: datetime
    nb_examens: int = 0

    model_config = {"from_attributes": True}


class MedecinCreateSchema(BaseModel):
    email: EmailStr
    nom: str = Field(min_length=2, max_length=255)
    password: str = Field(min_length=8, max_length=128)
    actif: bool = True


class MedecinUpdateSchema(BaseModel):
    email: EmailStr | None = None
    nom: str | None = Field(default=None, min_length=2, max_length=255)
    password: str | None = Field(default=None, min_length=8, max_length=128)
    actif: bool | None = None

File: frontend/src/types/medecin.ts
export interface Medecin {
  id: number
  email: string
  nom: string
  actif: boolean
  created_at: string
  nb_examens: number
}

export interface MedecinCreatePayload {
  email: string
  nom: string
  password: string
  actif?: boolean
}

export interface MedecinUpdatePayload {
  email?: string
  nom?: string
  password?: string
  actif?: boolean
}
